privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ToggleReportsDataStore.kt

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.privacy.dashboard.impl
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringSetPreferencesKey
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.common.utils.formatters.time.DatabaseDateFormatter
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.privacy.dashboard.impl.SharedPreferencesToggleReportsDataStore.Keys.TOGGLE_REPORTS_PROMPTS_DISMISSED
+import com.duckduckgo.privacy.dashboard.impl.SharedPreferencesToggleReportsDataStore.Keys.TOGGLE_REPORTS_SENT
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonClass
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.withContext
+
+interface ToggleReportsDataStore {
+
+    suspend fun getDismissLogicEnabled(): Boolean
+
+    suspend fun getPromptLimitLogicEnabled(): Boolean
+
+    suspend fun getDismissInterval(): Int
+
+    suspend fun getPromptInterval(): Int
+
+    suspend fun getMaxPromptCount(): Int
+
+    suspend fun insertTogglePromptDismiss()
+
+    suspend fun getReportsSent(): List<String>
+
+    suspend fun insertTogglePromptSend()
+
+    suspend fun getPromptsDismissed(): List<String>
+
+    suspend fun canPrompt(): Boolean
+}
+
+@JsonClass(generateAdapter = true)
```

Comment:
You're not using Moshi codegen to generate adapters, so this annotation can be removed.